samples/boards/intel_adsp/core_power_toggle/src/main.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/kernel.h>
+#include <zephyr/pm/pm.h>
+#include <zephyr/pm/state.h>
+#include <zephyr/logging/log.h>
+#include <soc.h>
+
+LOG_MODULE_REGISTER(power_sample, LOG_LEVEL_DBG);
+
+static inline bool cpu_is_primary(uint32_t core_id)
+{
+	return core_id == 0;
+}
+
+int cpu_enable_core(int id)
+{
+	/* only called from single core, no RMW lock */
+	__ASSERT_NO_MSG(cpu_is_primary(arch_proc_id()));
+	/*
+	 * This is an open-coded version of zephyr/kernel/smp.c
+	 * z_smp_start_cpu(). We do this, so we can use a customized
+	 * secondary_init() for SOF.
+	 */
```

Comment:
I think this comment is actual moot, Now with k_smp_cpu_start() interface, we no longer need to open-code functionality from Zephyr smp.c.